src/test/java/org/opensearch/neuralsearch/processor/factory/RerankProcessorFactoryTests.java

```diff
@@ -0,0 +1,190 @@
+import java.util.Locale;
+import java.util.Map;
+
+import org.junit.Before;
+import org.mockito.Mock;
+import org.opensearch.OpenSearchParseException;
+import org.opensearch.neuralsearch.ml.MLCommonsClientAccessor;
+import org.opensearch.neuralsearch.processor.rerank.MLOpenSearchRerankProcessor;
+import org.opensearch.neuralsearch.processor.rerank.RerankProcessor;
+import org.opensearch.neuralsearch.processor.rerank.RerankType;
+import org.opensearch.neuralsearch.processor.rerank.context.DocumentContextSourceFetcher;
+import org.opensearch.search.pipeline.Processor.PipelineContext;
+import org.opensearch.search.pipeline.SearchResponseProcessor;
+import org.opensearch.test.OpenSearchTestCase;
+
+public class RerankProcessorFactoryTests extends OpenSearchTestCase {
+
+    final String TAG = "default-tag";
+    final String DESC = "processor description";
+
+    private RerankProcessorFactory factory;
+
+    @Mock
+    private MLCommonsClientAccessor clientAccessor;
+
+    @Mock
+    private PipelineContext pipelineContext;
+
+    @Before
+    public void setup() {
+        pipelineContext = mock(PipelineContext.class);
+        clientAccessor = mock(MLCommonsClientAccessor.class);
+        factory = new RerankProcessorFactory(clientAccessor);
+    }
+
+    public void testRerankProcessorFactory_EmptyConfig_ThenFail() {
+        Map<String, Object> config = new HashMap<>(Map.of());
+        assertThrows(
+            "no rerank type found",
+            IllegalArgumentException.class,
+            () -> factory.create(Map.of(), TAG, DESC, false, config, pipelineContext)
+        );
+    }
+
+    public void testRerankProcessorFactory_NonExistentType_ThenFail() {
+        Map<String, Object> config = new HashMap<>(
+            Map.of("jpeo rvgh we iorgn", Map.of(MLOpenSearchRerankProcessor.MODEL_ID_FIELD, "model-id"))
+        );
+        assertThrows(
+            "no rerank type found",
+            IllegalArgumentException.class,
+            () -> factory.create(Map.of(), TAG, DESC, false, config, pipelineContext)
+        );
+    }
+
+    public void testRerankProcessorFactory_CrossEncoder_HappyPath() {
+        Map<String, Object> config = new HashMap<>(
+            Map.of(
+                RerankType.ML_OPENSEARCH.getLabel(),
+                new HashMap<>(Map.of(MLOpenSearchRerankProcessor.MODEL_ID_FIELD, "model-id")),
+                RerankProcessorFactory.CONTEXT_CONFIG_FIELD,
+                new HashMap<>(Map.of(DocumentContextSourceFetcher.NAME, new ArrayList<>(List.of("text_representation"))))
+            )
+        );
+        SearchResponseProcessor processor = factory.create(Map.of(), TAG, DESC, false, config, pipelineContext);
+        assert (processor instanceof RerankProcessor);
+        assert (processor instanceof MLOpenSearchRerankProcessor);
+        assert (processor.getType().equals(RerankProcessor.TYPE));
+    }
+
+    public void testRerankProcessorFactory_CrossEncoder_MessyConfig_ThenHappy() {
```

Comment:
our convention for method names is "testFeature_whenScenario_thenExpectedResult", you can put here something like: `testCrossEncoder_whenCorrectConfig_thenSuccessful`. Please change other method names accordingly